test/terraform_aws_s3_private_bucket_test.go

```diff
@@ -373,3 +373,78 @@ func TestTerraformAwsS3PrivateBucketNoLoggingBucket(t *testing.T) {
 	AssertS3BucketLoggingNotEnabled(t, awsRegion, testName)
 	AssertS3BucketPolicyContainsNonTLSDeny(t, awsRegion, testName)
 }
+
+func AssertS3BucketAnalyticsEnabled(t *testing.T, region string, bucketName string) {
+	err := AssertS3BucketAnalyticsEnabledE(t, region, bucketName)
+	require.NoError(t, err)
+}
+
+func AssertS3BucketAnalyticsEnabledE(t *testing.T, region string, bucketName string) error {
+	s3Client, err := aws.NewS3ClientE(t, region)
+
+	if err != nil {
+		return err
+	}
+
+	params := &s3.ListBucketAnalyticsConfigurationsInput{
+		Bucket: awssdk.String(bucketName),
+	}
+
+	bucketAnalytics, err := s3Client.ListBucketAnalyticsConfigurations(params)
+
+	if err != nil {
+		return err
+	}
+
+	analyticsEnabled := bucketAnalytics.IsTruncated
```

Comment:
I think this just checks for pagination in `ListBucketAnalyticsConfigurations`, not that Analytics is enabled. I think https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#S3.GetBucketAnalyticsConfiguration is what you are looking for.

Reply:
Aaah Thanks. That's what I had originally setup in [this commit](https://github.com/trussworks/terraform-aws-s3-private-bucket/pull/95/commits/63e16c559a8abd4d8155c9ecc9387dc5c964592b) but I lost my confidence and started plugging in other stuff. Makes sense - I'll change it back.